src/NuGet.Clients/NuGet.PackageManagement.UI/Actions/UIActionEngine.cs

```diff
@@ -392,18 +394,20 @@ private static string GenerateUpgradeReport(string projectName, string backupPat
                     {
                         // log rich info about added packages
                         addedPackages = results.SelectMany(result => result.Added)
-                            .Select(package => new Tuple<string, string>(package.Id, (package.Version == null ? "" : package.Version.ToNormalizedString())))
+                            .Select(package => (package.Id,
+                                                package.Version == null ? "" : package.Version.ToNormalizedString(),
+                                                package.Id.Equals(userAction.PackageId, StringComparison.OrdinalIgnoreCase) ? userAction.VersionRange : VersionRange.Parse(package.Version == null ? "" : package.Version.ToNormalizedString())))
                             .Distinct()
                             .ToList();
                         var addCount = addedPackages.Count;
 
                         //updated packages can have an old and a new id.
                         updatedPackagesOld = results.SelectMany(result => result.Updated)
-                            .Select(package => new Tuple<string, string>(package.Old.Id, (package.Old.Version == null ? "" : package.Old.Version.ToNormalizedString())))
+                            .Select(package => (package.Old.Id, (package.Old.Version == null ? "" : package.Old.Version.ToNormalizedString()), package.OldVersionRange))
```

Comment:
You probably need to look at the input packages only. 
You only care about the current installation right?